lib/faker/date.ex

```diff
@@ -12,23 +12,58 @@ defmodule Faker.Date do
       date_of_birth(1) #=> ~D[2015-12-06]
       date_of_birth(10..19) #=> ~D[2004-05-15]
   """
-  @spec date_of_birth() :: Date.t
+  @spec date_of_birth() :: Date.t()
   def date_of_birth(age_or_range \\ 18..99)
 
-  @spec date_of_birth(integer) :: Date.t
+  @spec date_of_birth(integer) :: Date.t()
   def date_of_birth(age) when is_integer(age) do
-    date_of_birth(%Range{first: age, last: age + 1})
+    {{year_now, month_now, day_now}, _time} = :calendar.local_time()
+
+    earliest_year = year_now - (age + 1)
+
+    potential_earliest_date = {earliest_year, month_now, day_now + 1}
+    potential_latest_date = {earliest_year + 1, month_now, day_now}
+
+    earliest_date =
+      case :calendar.valid_date(potential_earliest_date) do
+        true -> {earliest_year, month_now, day_now + 1}
+        false -> {earliest_year, 3, 1}
+      end
+
+    latest_date =
+      case :calendar.valid_date(potential_latest_date) do
+        true -> {earliest_year + 1, month_now, day_now}
+        false -> {earliest_year + 1, 2, 28}
+      end
+
+    earliest_as_seconds = :calendar.datetime_to_gregorian_seconds({earliest_date, {0, 0, 0}})
+    lastest_as_seconds = :calendar.datetime_to_gregorian_seconds({latest_date, {23, 59, 59}})
+
+    {chosen_date, _time} =
+      (lastest_as_seconds - earliest_as_seconds)
+      |> :rand.uniform()
+      |> Kernel.+(earliest_as_seconds)
+      |> :calendar.gregorian_seconds_to_datetime()
+
+    {:ok, result} = Date.from_erl(chosen_date)
+    result
   end
 
-  @spec date_of_birth(Range.t) :: Date.t
+  @spec date_of_birth(Range.t()) :: Date.t()
   def date_of_birth(%Range{first: first, last: last}) do
-    {{current_year, current_month, current_day}, _time} = :calendar.local_time()
-    random_month = :crypto.rand_uniform(1, 13)
-    random_day = :crypto.rand_uniform(1, 29)
-    already_aged_this_year = current_month > random_month || current_month == random_month && random_day >= current_day
-    random_year = current_year - :crypto.rand_uniform(first, last) + (if already_aged_this_year, do: 1, else: 0)
-    {:ok, date} = Date.new random_year, random_month, random_day
-    date
+    random_age =
+      cond do
+        first < last ->
+          first + :rand.uniform(last - first + 1) - 1
```

Comment:
I've been like "wtf" at this line for a solid 5 minutes but it actually checks out 👍

Reply:
If i look at it again now...
This could be simpler.
And will be, in a couple of minutes :)